### PR TITLE
Test suite for server interpreters

### PR DESCRIPTION
**Pull request by Yuzvin**
Wrote test suite for server interpreter and tested it using Akka-Http server.

Fixes #117

### CODE REVIEW

**Comment by julienrf** [comment]
Fixes #117

File: algebras/algebra/src/test/scala/endpoints/algebra/server/EndpointsTestSuite.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        val server = serveEndpoint(serverApi.smokeEndpoint, mockedResponse)
+        server.start()
+
+        val response  = sttp.get(uri"http://localhost:${server.port}/user/userId/description?name=name1&age=18").send()
+        assert(response.body.isRight)
+        assert(response.body.right.get == mockedResponse)
+        assert(response.code == 200)
+
+        server.stop()
+      }
```
Comment (julienrf) [inline algebras/algebra/src/test/scala/endpoints/algebra/server/EndpointsTestSuite.scala]: What do you think if, instead, `serveEndpoint` takes an additional function parameter that is called after the server is started and before it is stopped? Test writers wouldn’t have to manually take care of starting and stopping the server.

File: algebras/build.sbt [hunk 2]
```diff
@@ -8,6 +8,7 @@ val algebra =
       name := "endpoints-algebra",
       libraryDependencies ++= Seq(
         "com.github.tomakehurst" % "wiremock" % "2.6.0" % Test,
+        "com.softwaremill.sttp" %% "core" % "1.3.2" % Test,
```
Comment (julienrf) [inline algebras/build.sbt]: The CI failed with an error related to this dependency. Can you check that we don’t miss another dependency?

Comment (NK91) [reply algebras/build.sbt]: It seems `sttp` has dependencies to `akka-http 10.1.4` and `akka-stream 2.5.16`, maybe it's the reason of failure?

Comment (NK91) [reply algebras/build.sbt]: I think changing the version of the `sttp` to `1.1.13` can help.

**Pull request merged**